kolibri/plugins/user/middleware.py

```diff
@@ -0,0 +1,24 @@
+from django.core.urlresolvers import reverse
+from django.shortcuts import redirect
+from django.utils.deprecation import MiddlewareMixin
+
+from kolibri.core.auth.models import Facility
+from kolibri.core.auth.models import KolibriAnonymousUser
+
+
+SESSION_URL = reverse('session-list')
+USER_SIGNIN_URL = reverse("kolibri:user:user")
+
+
+class RedirectToSignInPageIfNoGuestAccessAndNoActiveSession(MiddlewareMixin):
+    """
+    Redirect to the sign in page if facility does not allow guest access and user
+    has not signed into the facility.
+    """
+
+    def process_request(self, request):
```

Comment:
We should also allow a few more urls like `set_language` and `admin`, etc url?

Reply:
can you confirm if `set_language` is blocked by the redirect?

Reply:
Confirming.